From fc45d80073c62f6b067a3227aa2a301b8b5e2336 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Tue, 11 Apr 2023 00:10:47 +0200
Subject: [PATCH] ARM: dts: stm32: add the arm-wdt support on STM32MP1

Add the support for arm-wdt for STM32MP13 and STM32MP15.

Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: If2e3f30caed2ea4589ff947be223c5e332687aaf
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 6 ++++++
 arch/arm/boot/dts/st/stm32mp151.dtsi | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp131.dtsi b/arch/arm/boot/dts/st/stm32mp131.dtsi
index ac90fcbf0c09..2dfc1b5f4628 100644
--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -29,6 +29,12 @@ arm-pmu {
 		interrupt-parent = <&intc>;
 	};
 
+	arm_wdt: watchdog {
+		compatible = "arm,smc-wdt";
+		arm,smc-id = <0xb200005a>;
+		status = "disabled";
+	};
+
 	firmware {
 		optee {
 			method = "smc";
diff --git a/arch/arm/boot/dts/st/stm32mp151.dtsi b/arch/arm/boot/dts/st/stm32mp151.dtsi
index aec7fa5ab5d8..aa72a52a1464 100644
--- a/arch/arm/boot/dts/st/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp151.dtsi
@@ -30,6 +30,12 @@ arm-pmu {
 		interrupt-parent = <&intc>;
 	};
 
+	arm_wdt: watchdog {
+		compatible = "arm,smc-wdt";
+		arm,smc-id = <0xb200005a>;
+		status = "disabled";
+	};
+
 	psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
-- 
2.39.5

